domain/src/main/java/edu/nextstep/camp/calculator/domain/Calculator.kt

```diff
@@ -0,0 +1,56 @@
+package edu.nextstep.camp.calculator.domain
+
+class Calculator {
+
+    fun evaluate(expression: String?): Int {
+        if (expression.isNullOrBlank())
+            throw IllegalArgumentException("null 또는 빈 공백 문자열은 수식이 아닙니다.")
+
+        return evaluateSymbols(splitToSymbols(expression))
```

Comment:
evaluateSymbols(splitToSymbols(expression))에 대한 함수 정의 순서를 evaluateSymbols → evaluateOne → splitToSymbols 순으로 작성하시면 좋을 것 같습니다.

‘클린코드 3장 함수’ 편에 보면

`코드는 위에서 아래로 이야기처럼 읽혀야 좋다. 한 함수 다음에는 추상화 수준이 한 단계 낮은 함수가 온다. 위에서 아래로 프로그램을 읽으면 함수 추상화 수준이 한 단계씩 낮아진다. 이것을 ‘내려가기 규칙’이라 부른다.`

라고 정의하고 있습니다.

작성하신 함수의 '의도'를 보자면 왜 splitToSymbols를 먼저 적으신지 이해는 갑니다. 제가 그런 의도가 있었다면 evaluateSymbols(splitToSymbols(…)) 형태가 아니라 splitToSymbols 함수의 결과를 저장하는 변수 선언을 evaluateSymbols 함수 위에 작성했을것 같습니다.

```suggestion

        val symbolList = splitToSymbols(expression)
        return evaluateSymbols(symbolList)
```

이렇게 코드를 변경한다면 이후 정의한 함수 순서대로 읽어 내려가게 될것 같습니다. 😘